homeassistant/components/broadlink/time.py

```diff
@@ -0,0 +1,70 @@
+"""Support for Broadlink device time."""
+
+from __future__ import annotations
+
+from datetime import time
+from typing import Any
+
+from homeassistant.components.time import TimeEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from . import BroadlinkDevice
+from .const import DOMAIN
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink time."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+    async_add_entities([BroadlinkTime(device)])
+
+
+class BroadlinkTime(BroadlinkEntity, TimeEntity):
+    """Representation of a Broadlink device time."""
+
+    _attr_has_entity_name = True
+    _attr_native_value: time | None = None
+
+    def __init__(self, device: BroadlinkDevice) -> None:
+        """Initialize the sensor."""
+        super().__init__(device)
+
+        self._attr_unique_id = f"{device.unique_id}-device-time"
+
+    def _update_state(self, data: dict[str, Any]) -> None:
+        """Update the state of the entity."""
+        if data is None or "hour" not in data or "min" not in data or "sec" not in data:
+            self._attr_native_value = None
+        else:
+            self._attr_native_value = time(
+                hour=data["hour"],
+                minute=data["min"],
+                second=data["sec"],
+                tzinfo=dt_util.get_default_time_zone(),
+            )
+
+    async def async_set_value(self, value: time) -> None:
+        """Change the value."""
+        if self._coordinator.data is None or "dayofweek" not in self._coordinator.data:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="request_failed_device_not_connected",
+            )
```

Comment:
Shouldn't the entity be unavailable in that moment? Or are there other moments when this can happen

Reply:
I had the direct service call in mind. And at least during testing the device when Homeassistant first connected to the device (after bootup) it was available but no time was displayed, meaning the coordinator had to data yet.

Something I want to investigate and possible fix in a separate PR

Reply:
But why is that the case? you're now raising an issue that the device is not connected, but from your response I also guess that it already was connected, so what should the user do to get the data?

Reply:
Removed